### PR TITLE
pickle to msgpack changes

### PR DESCRIPTION
**Pull request by rjgildea**
This makes the necessary changes to xia2 now that msgpack is the default reflection file format in dials (dials/dials#743). All tests pass with `$ pytest --regression-full -n auto` apart from 2 tests which fail tolerance tests (I think these were failing before the msgpack changes were merged in to dials master).

### CODE REVIEW

**Comment by graeme-winter** [comment]
Wilco

File: command_line/report.py [hunk 1]
```diff
@@ -1160,14 +1160,13 @@ def z_vs_I_over_sigma(self):
...
+        import iotbx.mtz
 
         self.unmerged_mtz = unmerged_mtz
         self.params = params
 
-        reader = any_reflection_file(unmerged_mtz)
-        assert reader.file_type() == "ccp4_mtz"
-        arrays = reader.as_miller_arrays(merge_equivalents=False)
+        self.mtz_object = iotbx.mtz.object(unmerged_mtz)
+        arrays = self.mtz_object.as_miller_arrays(merge_equivalents=False, auto_anomalous=True)
```
Comment (Anthchirp) [inline command_line/report.py]: Changes in this file are unrelated?

File: Modules/Scaler/rebatch.py [hunk 2]
```diff
@@ -122,12 +141,28 @@ def rebatch(
...
+
     new_batches = (min(batches) + offset, max(batches) + offset)
 
+    t9 = time.time()
+
+    print(t1 - t0)
+    print(t2 - t1)
+    print(t3 - t2)
+    print(t4 - t3)
+    print(t5 - t4)
```
Comment (Anthchirp) [inline Modules/Scaler/rebatch.py]: accidental add?

File: command_line/report.py (hunk 1 above)
Comment (rjgildea) [reply command_line/report.py]: It turns out pull requests are useful 😉
